sklearn/ensemble/forest.py

```diff
@@ -678,14 +691,18 @@ def predict(self, X):
         # Assign chunk of trees to jobs
         n_jobs, _, _ = _partition_estimators(self.n_estimators, self.n_jobs)
 
+        # avoid storing the output of every estimator by summing them in `out`
+        if self.n_outputs_ > 1:
+            y_hat = np.zeros((X.shape[0], self.n_outputs_), dtype=np.float64)
+        else:
+            y_hat = np.zeros((X.shape[0]), dtype=np.float64)
+
         # Parallel loop
-        all_y_hat = Parallel(n_jobs=n_jobs, verbose=self.verbose,
-                             backend="threading")(
-            delayed(parallel_helper)(e, 'predict', X, check_input=False)
```

Comment:
Given that this now works without `parallel_helper` due to the threading backend, I suspect we can remove all uses of `parallel_helper` in random forests. Another PR is welcome.